src/evict/evict_lru.c

```diff
@@ -2096,9 +2100,11 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         }
 
         /* Skip pages we don't want. */
-        want_page = (F_ISSET(cache, WT_CACHE_EVICT_CLEAN) && !modified) ||
+        want_page = (F_ISSET(cache, WT_CACHE_EVICT_CLEAN) && !F_ISSET(btree, WT_BTREE_IN_MEMORY) &&
+                      !modified) ||
           (F_ISSET(cache, WT_CACHE_EVICT_DIRTY) && modified) ||
-          (F_ISSET(cache, WT_CACHE_EVICT_UPDATES) && page->modify != NULL);
+          (F_ISSET(cache, WT_CACHE_EVICT_UPDATES) && !F_ISSET(btree, WT_BTREE_IN_MEMORY) &&
+            page->modify != NULL);
```

Comment:
I'm not super familiar with eviction and this change is a little hard to follow - a comment in the code wouldn't be astray.